### PR TITLE
Fix issue where map updates don't take effect in Flutter v3.0.0

### PR DESCRIPTION
**Pull request by blasten**
Android platform views that are displayed as a texture after Flutter v3.0.0.
require that the view hierarchy is notified after all drawing operations have been flushed.
Since the gmscore GL renderer doesn't use standard Android views, and instead uses TextureView,
we notify the view hierarchy by invalidating the view.
Unfortunately, when OnMapLoadedCallback is fired, the texture may not have been updated yet.
To workaround this limitation, wait two frames.
This ensures that at least the frame budget (16.66ms at 60hz) have passed since the
drawing operation was issued.

Ideally, `google_maps_flutter` is rendered a texture, so there's a single mode for platform views, and there's fewer surprises when combined with other platform views like WebView or AdMobView.

I will work on updating the documentation.
Fixes https://github.com/flutter/flutter/issues/103686

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java
Comment (blasten) [reply packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java:272]: something seems special about updating markers. I have not been able to reproduce this issue by triggering the other code paths.

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:3]: Nit: I would add "on Android" here.

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java:297]: Since this is a block of specialized, non-trivial logic, I would definitely suggest putting it in a helper function. Then the comment about how it works can go there, and this can become just a call to that helper with a comment explaining why it's only done in `markers#update` (since it's definitely non-obvious).

That will also make it trivial to add this to other codepaths if we discover other edge cases later that have this same issue.

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md
Comment (blasten) [reply packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:3]: done

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java
Comment (blasten) [reply packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java]: done

**Review by GaryQian — APPROVED** [review]
RSLGTM, new to this codebase, so defer to Stuart for final sign off.

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java:153]: Don't we actually need to restart the callback counter if a new invalidate request comes in? Otherwise we won't actually wait 2 frames after the last invalidation.

Comment (blasten) [reply packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java]: I changed it to `loadedCallbackPending`. Note that `setOnMapLoadedCallback` is fired when the map has finished rendering, so gmscore has a batching mechanism.

**Comment by blasten** [comment]
This is a temporary fix for the stable branch. I'm going to see what can be done in the engine to remove this issue. I know this is a special case with any view that uses SurfaceView or TextureView and does its own drawing.

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java:162]: Are we guaranteed that this is never called syncronously for updates we make? If not, then we need to make `invalidateMapIfNeeded` calls *first* in all of our method handers, not last, to make sure we get this, right?

**Review by stuartmorgan — APPROVED** [review]
LGTM modulo an ordering question.

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java
Comment (blasten) [reply packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java:162]: done

**Pull request merged**
